cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/QueryResolver.scala

```diff
@@ -192,7 +192,10 @@ object QueryResolver {
     val field = fieldMap.get(name)
     field match {
       case Some(f) => f
-      case _ => throw FieldNotFound(name)
+      case _ => fieldMap.get(name.split("\\.", 2).head) match {
+          case Some(f) if (f.dataType == DataType.Json && name.contains(".")) => new JsonField(name, false) // name has the format of 'fieldName.subFieldName'
```

Comment:
A request's fieldName might not be in the fieldMap, e.g.: 
Field: {
      "name" : "user.id",
      "isOptional" : false,
      "datatype" : "Json"
 }, but "user" is the keyword in fieldMap, so we simply create a new JsonField, with name equals to "user.id".